spec/api-system-preferences-spec.js

```diff
@@ -35,6 +35,35 @@ describe('systemPreferences module', () => {
     })
   })
 
+  describe('systemPreferences.registerDefaults(defaults)', () => {
+    before(function () {
+      if (process.platform !== 'darwin') this.skip()
+    })
+
+    it('registers defaults', () => {
+      const userDefaults = {
+        'one': 'ONE',
+        'two': 'TWO',
+        'three': 'THREE'
+      }
+
+      systemPreferences.registerDefaults(userDefaults)
+
+      for (const [key, expectedValue] of Object.entries(userDefaults)) {
+        const actualValue = systemPreferences.getUserDefault(key, 'string')
+        assert.equal(actualValue, expectedValue)
+      }
+    })
+
+    it('throws when bad defaults are passed', () => {
+      const badDefaults1 = { 'one': null }
```

Comment:
i think we can reasonably expect clients to be responsible for passing in a dictionary, so the method now just catches for invalid keys/values within the dict

Reply:
> can reasonably expect clients to be responsible for passing in a dictionary

We can, but we should not =)
We should be ready to get literally anything, and at least do no crash on invalid values.
And provide a meaningful error message if possible.

Reply:
ok sounds good! i'll improve error handling this morning :)